plugins/discovery/discovery_test.go

```diff
@@ -1346,6 +1347,40 @@ func TestReconfigureWithUpdates(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Unexpected error %v", err)
 	}
+
+	// check that not setting persistence_directory doesn't remove boot config
+	updatedBundle = makeDataBundle(13, `
+		{
+			"config": {}
+		}
+	`)
+
+	err = disco.reconfigure(ctx, download.Update{Bundle: updatedBundle})
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	if manager.Config.PersistenceDirectory == nil {
+		t.Fatal("Erased persistence directory configuration")
+	}
+
+	// update persistence directory
+	updatedBundle = makeDataBundle(14, `
+		{
+			"config": {
+				"persistence_directory": "test"
```

Comment:
Nit: Let's change the directory name to something other than `test` as the original config has the same persistence directory.